daml-foundations/daml-ghc/daml-stdlib-src/DA/Text.daml

```diff
@@ -187,6 +187,13 @@ splitOn (explode -> s) (explode -> t) =
   let fixit = if L.isSuffixOf s t then (++ [""]) else identity
   in fixit $ implode <$> L.repeatedly (fmap (L.dropPrefix s) . L.breakOn s) t
 
+-- | Split a text before a given position so that
+-- `length (fst (splitAt n t)) == n`
```

Comment:
This comment isn't quite accurate - all the function guarantees is that the length of the fst result is a max of n and the length of the string. It can definitely be shorter.